src/Compilation.zig

```diff
@@ -646,6 +646,12 @@ pub const InitOptions = struct {
     framework_dirs: []const []const u8 = &[0][]const u8{},
     frameworks: []const []const u8 = &[0][]const u8{},
     system_libs: []const []const u8 = &[0][]const u8{},
+    /// These correspond to the WASI libc emulated subcomponents including:
+    /// * process clocks
+    /// * getpid
+    /// * mman
+    /// * signal
+    wasi_emulated_libs: []const []const u8 = &[0][]const u8{},
```

Comment:
These should be enum items and not strings since the strings are checked by the CLI, right?

Reply:
The strings correspond to the names of required emulated WASI libc subcomponents imported as libraries on the linker line, e.g., `-lwasi-emulated-process-clocks` would result in us saving `wasi-emulated-process-clocks` in `wasi_emulated_libs` slice. We could store a slice of enums too, but we still need to get the name of the matching pre-compiled lib using `comp.get_libc_crt_file(arena, "libwasi-emulated-process-clocks.a")`. I'm fine either way TBH, and if you feel that matching strings to enum variants in `main.zig` makes more sense, I'm happy to make the necessary tweaks.